docs/modules/ROOT/partials/getting-started/proc-managing-artifacts-using-maven-plugin.adoc

```diff
@@ -12,67 +12,111 @@
-      </artifacts>
-    </configuration>
-  </execution>
-</executions>
+  <groupId>io.apicurio</groupId>
+  <artifactId>apicurio-registry-maven-plugin</artifactId>
+  <version>${registry.version}</version>
+  <executions>
+    <execution>
+      <phase>generate-sources</phase>
+      <goals>
+        <goal>register</goal> <1>
+      </goals>
+      <configuration>
+        <registryUrl>http://my-cluster-service-registry-myproject.example.com/api</registryUrl> <2>
+        <artifactType>AVRO</artifactType>
+        <artifacts>
+          <schema1>${project.basedir}/schemas/schema1.avsc</schema1> <3>
+        </artifacts>
+      </configuration>
+    </execution>
+  </executions>
 </plugin>
 ----
-+
 <1> Specify `register` as the execution goal to upload an artifact to the registry.
 <2> You must specify the {registry} URL with the `/api` endpoint.
-<3> You can upload multiple artifacts using the artifact ID and location. 
+<3> You can upload multiple artifacts using the artifact ID and location.
+
+== Downloading an artifact using the Maven plug-in
+You can also use the Maven plug-in to download artifacts from {registry}.  This is often useful, for example, when
+generating code from a registered schema.
+
+The following example shows downloading a single schema by its artifact ID.
 
-. You can also update your Maven `pom.xml` file to download a previously registered artifact from {registry}:
-+
 [source,xml]
 ----
 <plugin>
 <groupId>io.apicurio</groupId>
 <artifactId>apicurio-registry-maven-plugin</artifactId>
 <version>${registry.version}</version>
-<executions>   
+<executions>
   <execution>
-    <phase>generate-sources</phase> 
+    <phase>generate-sources</phase>
     <goals>
       <goal>download</goal> <1>
     </goals>
     <configuration>
       <registryUrl>http://my-cluster-service-registry-myproject.example.com/api</registryUrl> <2>
-          <ids>
-            <param1>schema1</param1> <3>
-          </ids>
+      <ids>
+        <param1>schema1</param1> <3>
+      </ids>
+      <artifactExtension>.avsc</artifactExtension> <4>
       <outputDirectory>${project.build.directory}</outputDirectory>
    </configuration>
  </execution>
-</executions>   
+</executions>
 </plugin>
 ----
-+
 <1> Specify `download` as the execution goal.
 <2> You must specify the {registry} URL with the `/api` endpoint.
 <3> You can download multiple artifacts to a specified directory using the artifact ID.
+<4> The plug-in will automatically (try to) choose an appropriate file extension, but you can override it using `<artifactExtension>`.
+
+== Testing an artifact
+You may want to simply verify that an artifact can be registered without actually making any changes.  This is most
+often useful when rules have been configured in {registry}.  Testing the artifact will result in a failure if the
+artifact content violates any of the configured rules.
+
+NOTE: No content will be added to {registry} even if it passes all of the configured rules.
```

Comment:
Suggest "even if your test passes.." or "even if your artifact content passes" to be more clear?